src/app/maptools/qgsmaptoolshapecircleabstract.cpp

```diff
@@ -29,7 +32,19 @@ void QgsMapToolShapeCircleAbstract::addCircleToParentTool()
 
   mParentTool->clearCurve();
 
-  std::unique_ptr<QgsCircularString> lineString( mCircle.toCircularString() );
-
-  mParentTool->addCurve( lineString.release() );
+  // Check whether to draw the circle as a polygon or a circular string
+  const QgsCoordinateReferenceSystem layerCrs = mParentTool->layer()->crs();
```

Comment:
Note to self -- double check if layer can be nullptr here (I think it can)

Reply:
@YoannQDQ before this gets merged to LTR in https://github.com/qgis/QGIS/pull/51945/files, can you add a safety check for the nullptr layer issue?

Reply:
@nyalldawson Do you want me to close #51945 and manually open another backport PR? Not quite sure how to do it otherwise.

Reply:
@YoannQDQ if you open a pr against master for just the layer nullptr change, then I'll merge the existing backport and then immediately the new backport after.

Reply:
Done